src/components/Accordion/Accordion.test.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { render, fireEvent,screen } from '@testing-library/react';
+import Accordion from "./Accordion";
+
+import AccordionHeader from "./AccordionHeader";
+import AccordionDetail from "./AccordionDetail";
+
+describe('Accordion', () => {
+
+  it('should match snapshot', () => {
+    const { asFragment } = render(
+    <Accordion>
+    <AccordionHeader >
+      <p>hello</p>
+    </AccordionHeader>
+    <AccordionDetail>
+      <p>New Accoirdon detail helloooo</p>
+    </AccordionDetail>
+  </Accordion>)
+  expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should render items', () => {
+    const headerSec = 'HelloFromHeader'
+    const detailSec = 'HelloFromDetail'
+    render(
+      <Accordion>
+      <AccordionHeader >
+        <p>{headerSec}</p>
+      </AccordionHeader>
+      <AccordionDetail>
+        <p>{detailSec}</p>
+      </AccordionDetail>
+    </Accordion>)
+    const headerEl = screen.queryByText(headerSec)
+    const detailEl = screen.queryByText(detailSec)
+    expect(headerEl).toBeInTheDocument();
+    expect(detailEl).toBeInTheDocument();
+  });
+
+  it('should open when the expanded props passed', () => {
+    const headerSec = 'HelloFromHeader'
+    const detailSec = 'HelloFromDetail'
+    render(
+      <Accordion expanded>
+      <AccordionHeader >
+        <p>{headerSec}</p>
+      </AccordionHeader>
+      <AccordionDetail>
+        <p>{detailSec}</p>
+      </AccordionDetail>
+    </Accordion>)
+    const activeDiv = document.querySelector('.accordion-container.active')
+    expect(activeDiv).toHaveClass('accordion-container active')
+  })
+
+  it('should open when the header clicked', () => {
+    const headerSec = 'HelloFromHeader'
+    const detailSec = 'HelloFromDetail'
+    render(
+      <Accordion>
+      <AccordionHeader >
+        <p>{headerSec}</p>
+      </AccordionHeader>
+      <AccordionDetail>
+        <p>{detailSec}</p>
+      </AccordionDetail>
+    </Accordion>)
+    const headerEl = screen.queryByText(headerSec) as HTMLDivElement
+    fireEvent.click(headerEl)
```

Comment:
test before click

Reply:
Should be good. Accordion element is checked to see if it does not have the 'active' class before the click. Then, it is checked after the click to see if it has the 'active' class